kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -463,168 +517,90 @@ define([
-                div({ class: 'modal-content' }, [
-                    div({ class: 'modal-header' }, [
-                        button(
-                            {
-                                type: 'button',
-                                class: 'close',
-                                dataDismiss: 'modal',
-                                ariaLabel: cancelLabel,
-                            },
-                            [span({ ariaHidden: 'true' }, '&times;')]
-                        ),
-                        span({ class: 'modal-title kb-title' }, title),
-                    ]),
-                    div({ class: 'modal-body' }, [content]),
-                    div(
-                        { class: 'modal-footer' },
-                        buttons
-                            .map((btn) => {
-                                return button(
-                                    {
-                                        type: 'button',
-                                        class: 'btn btn-' + (btn.type || 'default'),
-                                        dataElement: btn.action,
-                                    },
-                                    btn.label
-                                );
-                            })
-                            .concat([
-                                button(
-                                    {
-                                        type: 'button',
-                                        class: 'btn btn-default',
-                                        dataDismiss: 'modal',
-                                        dataElement: 'cancel',
-                                    },
-                                    cancelLabel
-                                ),
-                            ])
-                    ),
-                ]),
-            ]),
-        ]);
+        return _renderModal({
+            title: arg.title,
+            body: arg.body,
+            includeCancel: arg.cancelLabel || 'Cancel',
+            buttons: renderedButtons,
+            style: style,
+        });
     }
 
-    function showDialog(args) {
-        args.buttons = args.buttons || [];
-        const dialog = renderDialog(
-                args.title,
-                args.body,
-                args.cancelLabel || 'Cancel',
-                args.buttons,
-                args.options
-            ),
-            dialogId = html.genId(),
-            confirmNode = document.createElement('div');
-        let kbaseNode, modalNode;
-
-        confirmNode.id = dialogId;
-        confirmNode.innerHTML = dialog;
-
-        // top level element for kbase usage
-        kbaseNode = document.querySelector('[data-element="kbase"]');
-        if (!kbaseNode) {
-            kbaseNode = document.createElement('div');
-            kbaseNode.setAttribute('data-element', 'kbase');
-            document.body.appendChild(kbaseNode);
-        }
+    function showDialog(arg) {
```

Comment:
Of course, there is also another similar but not quite the same function that acts in a different way... I tried to unify the functionality with that of the other `show.+Dialog` functions without making the construction functions too tricksy.